47.md

```diff
@@ -126,6 +126,189 @@ Response:
+{
+    "result_type": "pay_keysend",
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+    }
+}
+```
+
+### `make_invoice`
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### `lookup_invoice`
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "lookup_invoice",
+    "result": {
+        "invoice": "string", // encoded invoice
+        "description": "string",
+        "description_hash": "string",
+        "preimage": "string",
+        "payment_hash": "string",
+        "amount": 123, // value in msats
+        "fees_paid": 123, // value in msats
+        "settled_at": unixtimestamp,
+        "metadata": {} // generic metadata that can be used to add things like zap/boostagram details for a payer name/comment/etc.
+    }
+}
+```
+
+
+### `list_transactions`
+
+Request:
+```jsonc
+{
+    "method": "list_transactions",
+    "params": {
+        "from": 1693876973, // starting timestamp in seconds since epoch, optional
+        "until": 1693876973, // ending timestamp in seconds since epoch, optional
+        "limit": 10, // maximum number of invoices to return, optional
+        "offset": 0, // offset of the first invoice to return, optional
+        "unpaid": true, // include unpaid invoices, optional, default false
+        "type": "incoming", // "incoming" for invoices, "outgoing" for payments, undefined for both
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "list_transactions",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "transactions": [
+            {
+               "type": "incoming", // "incoming" for invoices, "outgoing" for payments
+               "invoice": "string", // encoded invoice
+               "description": "string",
+               "description_hash": "string",
+               "preimage": "string",
+               "payment_hash": "string",
+               "amount": 123, // value in msats
+               "fees_paid": 123, // value in msats
+               "settled_at": unixtimestamp,
```

Comment:
missing a `created_at` timestamp

Reply:
Maybe `expires_at` too. (although both could be decoded from the invoice, it is nice to be available without having to decode first)

Reply:
We have a lot of info that can already be read from decoding the invoice. Probably fine to add these too